lightstep/instrument.py

```diff
@@ -177,12 +178,27 @@ def _flush_worker(self, connection):
 
     def _construct_report_request(self):
         """Construct a report request."""
+        report = None
         with self._mutex:
             report = ttypes.ReportRequest(self._runtime, self._span_records,
                                           self._log_records)
             self._span_records = []
             self._log_records = []
-            return report
+
+        for log in report.log_records:
+            if log.stable_name is not None:
+                log.stable_name = str(log.stable_name)
+            if log.payload_json is not None:
```

Comment:
Probably easier IRL....so the encoding on logging (rather than reporting) *does* have a significant performance impact? If it's not a clear performance necessity, I like "encoding early" as that seems like prone to hard-to-identify errors; if it is a performance necessity, well, then, it's a necessity so let's do the encoding at reporting time.

In either case, it might be nice to have a code comment explaining the conscious choice to do the encoding here (since the code alone isn't sufficient to convey that).